src/main/java/frc/robot/subsystems/intake/IntakeIOReal.java

```diff
@@ -19,6 +23,25 @@ public IntakeIOReal() {
 
   public void updateInputs(IntakeIOInputs inputs) {
     inputs.intakeAppliedVolts = motor.getMotorVoltage().getValueAsDouble();
+    inputs.gamePieceLocation = getGamePieceLocation();
+    inputs.hasGamePiece = checkForGamePiece();
+    // inputs.lastMeasurementTime = CANRANGE.getMeasurementTime().getValueAsDouble();
+  }
+
+  // gets coral location relative to the center of intake. gives a negative on the left side,
+  // positive value
+  // if on the right side of the intake
+  private double getGamePieceLocation() {
+    if (checkForGamePiece() == true) {
+      return CANRANGE.getDistance().getValueAsDouble() + coralRadius - canrangeOffset;
+    } else return 0.0;
+  }
+
+  private boolean checkForGamePiece() {
+    if (CANRANGE.getDistance().getValueAsDouble() > 0.4) { 
```

Comment:
Good start. There is also the case of intaking a piece sideways that you'll need to deal with.

Reply:
Not going to fix in this change